receiver/db.py

```
@@ -1084,6 +1085,17 @@ def run_retention_cleanup(self, general_days: int = 60, dns_days: int = 10,
                         result['deleted_so_far'], result['dns_deleted'],
                         result['non_dns_deleted'], general_days, dns_days)
+            # Run VACUUM ANALYZE after bulk deletes so dead tuples are reclaimed
+            # immediately, not left for autovacuum to find on its next cycle.
+            # VACUUM cannot run inside a transaction — requires autocommit.
+            try:
+                with self.get_conn() as conn:
+                    conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+                    with conn.cursor() as cur:
+                        cur.execute("VACUUM ANALYZE logs")
+                logger.info("Post-cleanup VACUUM ANALYZE complete")
+            except Exception as exc:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)